src/Symfony/Component/Translation/Translatable.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation;
+
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+// Load the global t() function
+require_once __DIR__.'/Resources/functions/translatable.php';
+
+/**
+ * @author Nate Wiebe <nate@northern.co>
+ */
+final class Translatable
+{
+    private $message;
+    private $parameters;
+    private $domain;
+
+    public function __construct(string $message, array $parameters = [], string $domain = 'messages')
+    {
+        $this->message = $message;
+        $this->parameters = $parameters;
+        $this->domain = $domain;
+    }
+
+    public function getMessage(): string
+    {
+        return $this->message;
+    }
+
+    public function getParameters(): array
+    {
+        return $this->parameters;
+    }
+
+    public function getDomain(): string
+    {
+        return $this->domain;
+    }
+
+    public static function trans(TranslatorInterface $translator, self $translatable, ?string $locale = null): string
```

Comment:
I don't get why this method is useful. Ideally we would like to be able to do `$translator->trans($translatable)` (which would be a BC break), but I fail to see how this method helps.

Reply:
It was specifically to avoid introducing a BC break. We can switch back to that, but if we do, what would be the best way to introduce this code?

Reply:
Ok, so IIUC, this is just a shortcut that people can use to translate a Translatable. My question is more when/where people would use this method.

Reply:
Essentially. The Twig trans function handles the `Translatable` object automatically here: https://github.com/symfony/symfony/pull/37670/files#diff-8159ed6f6b7dbc087fbc19d5ab20e98aR111

To translate the object in php without twig, you'd have to do something similar to that which could become a bit messy if you have it multiple places within a project. This function handles most of that logic for you instead.

Reply:
Just to provide an example, we'll likely add a translatable exception that we can use for both logging and/or displaying to the user. When we catch, we set a flash message/pass to the view, but could also then log something like:

```php
$logger->log(Translatable::trans($translator, $e->getTranslatable(), 'en'));
```